tests/equality/equality.cpp

```diff
@@ -0,0 +1,1034 @@
+{
+    libcellml::VariablePtr v1 = libcellml::Variable::create("variableA");
+    libcellml::VariablePtr v2 = libcellml::Variable::create("variableB");
+
+    EXPECT_NE(v1, v2);
+    EXPECT_FALSE(v1->equal(v2));
+    EXPECT_FALSE(v2->equal(v1));
+}
+
+TEST(Equality, variableNotEqualToUnits)
+{
+    libcellml::VariablePtr v1 = libcellml::Variable::create("variable");
+    libcellml::UnitsPtr u1 = libcellml::Units::create("variable");
+
+    EXPECT_FALSE(v1->equal(u1));
+    EXPECT_FALSE(v1->equal(u1));
+}
+
+TEST(Equality, variableEqualById)
+{
+    libcellml::VariablePtr v1 = libcellml::Variable::create("variable");
+    libcellml::VariablePtr v2 = libcellml::Variable::create("variable");
+
+    v1->setId("lefisjeij");
+    v2->setId("lefisjeij");
+
+    EXPECT_TRUE(v1->equal(v2));
+    EXPECT_TRUE(v2->equal(v1));
+}
+
+TEST(Equality, variableNotEqualById)
+{
+    libcellml::VariablePtr v1 = libcellml::Variable::create("variable");
+    libcellml::VariablePtr v2 = libcellml::Variable::create("variable");
+
+    v1->setId("eijei3009");
+    v2->setId("lefisjeij");
+
+    EXPECT_FALSE(v1->equal(v2));
+    EXPECT_FALSE(v2->equal(v1));
+}
+
+TEST(Equality, variableEqualAllTestedConditionsVariant1)
+{
+    libcellml::VariablePtr v1 = libcellml::Variable::create();
+    libcellml::VariablePtr v2 = libcellml::Variable::create();
+
+    v1->setName("variable");
+    v2->setName("variable");
+
+    v1->setInitialValue(3.0);
+    v2->setInitialValue("3");
+
+    v1->setInterfaceType(libcellml::Variable::InterfaceType::PUBLIC);
+    v2->setInterfaceType(libcellml::Variable::InterfaceType::PUBLIC);
+
+    v1->setUnits("ampere");
+    v2->setUnits("ampere");
+
+    EXPECT_TRUE(v1->equal(v2));
+    EXPECT_TRUE(v2->equal(v1));
+}
+
+TEST(Equality, variableEqualAllTestedConditionsVariant2)
+{
+    libcellml::VariablePtr v1 = libcellml::Variable::create("variable");
+    libcellml::VariablePtr v2 = libcellml::Variable::create("variable");
+
+    libcellml::VariablePtr v3 = libcellml::Variable::create("initialValue");
+    libcellml::VariablePtr v4 = libcellml::Variable::create("initialValue");
+
+    libcellml::UnitsPtr u1 = libcellml::Units::create("m_per_s");
+    libcellml::UnitsPtr u2 = libcellml::Units::create("m_per_s");
+
+    v1->setInitialValue(v3);
+    v2->setInitialValue(v4);
+
+    v1->setInterfaceType(libcellml::Variable::InterfaceType::PUBLIC);
+    v2->setInterfaceType(libcellml::Variable::InterfaceType::PUBLIC);
+
+    v1->setUnits(u1);
+    v2->setUnits(u2);
+
+    EXPECT_TRUE(v1->equal(v2));
+    EXPECT_TRUE(v2->equal(v1));
+}
+
+TEST(Equality, variableNotEqualByUnits)
+{
+    libcellml::VariablePtr v1 = libcellml::Variable::create("variable");
+    libcellml::VariablePtr v2 = libcellml::Variable::create("variable");
+
+    libcellml::UnitsPtr u1 = libcellml::Units::create("m_per_s");
+
+    v1->setUnits(u1);
+
+    EXPECT_FALSE(v1->equal(v2));
+    EXPECT_FALSE(v2->equal(v1));
+}
+
+TEST(Equality, variableNotEqualByUnitsOther)
+{
```

Comment:
Now that you have both directions of testing equality in the test above, this one isn't required?

Reply:
67ce1b8